src/traits.rs

```diff
@@ -687,3 +687,403 @@ pub trait SuffStat<X> {
     /// Combine sufficient statistics
     fn merge(&mut self, other: Self);
 }
+
+/// Trait for distributions that can be shifted by a constant value
+pub trait Shiftable {
+    type Output;
+    type Error;
+
+    fn shifted(self, dx: f64) -> Result<Self::Output, Self::Error>
+    where
+        Self: Sized;
+
+    fn shifted_unchecked(self, dx: f64) -> Self::Output
+    where
+        Self: Sized;
+}
+
+/// Macro to implement Shiftable for a distribution type
+///
+/// This macro automatically implements the Shiftable trait for a given type,
+/// using the default Shifted<T> as the Output type.
+#[macro_export]
+macro_rules! impl_shiftable {
+    // Simple case for non-generic types
+    ($type:ty) => {
+        use $crate::prelude::Shifted;
+        use $crate::prelude::ShiftedError;
+
+        impl Shiftable for $type {
+            type Output = Shifted<Self>;
+            type Error = ShiftedError;
+
+            fn shifted(self, dx: f64) -> Result<Self::Output, Self::Error>
+            where
+                Self: Sized,
+            {
+                Shifted::new(self, dx)
+            }
+
+            fn shifted_unchecked(self, dx: f64) -> Self::Output
+            where
+                Self: Sized,
+            {
+                Shifted::new_unchecked(self, dx)
+            }
+        }
+    };
+}
+
```

Comment:
Since `proptest` isn't included outside dev dependencies, I think the rest of this file should be wrapped in an `#[cfg(test)]` block.